code/game/objects/items/wayfinding.dm

```diff
@@ -0,0 +1,77 @@
+/obj/machinery/pinpointer_dispenser
+	name = "wayfinding pinpointer dispenser"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "ticketmachine"
+	desc = "Having trouble finding your way? This machine dispenses pinpointers that point to common locations."
+	density = FALSE
+	layer = HIGH_OBJ_LAYER
+	pixel_y = 32
+	var/list/obj/item/pinpointer/wayfinding/pinpointers = list()
+
+/obj/machinery/pinpointer_dispenser/attack_hand(mob/living/carbon/user)
+	if(user.name in pinpointers)
+		to_chat(user, "<span class='warning'>There's already a pinpointer registered to [user.name]!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You take a pinpointer from [src].</span>")
+
+	var/obj/item/pinpointer/wayfinding/P = new /obj/item/pinpointer/wayfinding(get_turf(src))
+	user.put_in_hands(P)
+	P.owner = user.name
+	pinpointers[user.name] = P
+
+/obj/item/pinpointer/wayfinding //For new players or new stations to help players find their way around
+	name = "wayfinding pinpointer"
+	desc = "A handheld tracking device that points to useful places."
+	icon_state = "pinpointer_crew"
+	var/owner = null
+	var/list/beacons = list()
+
+/obj/item/pinpointer/wayfinding/attack_self(mob/living/user)
+	if(active)
+		toggle_on()
+		user.visible_message("<span class='notice'>[user] deactivates [user.p_their()] pinpointer.</span>", "<span class='notice'>You deactivate your pinpointer.</span>")
```

Comment:
This wont show the user anything. Only blind_message shows when the object is inside something, in this case inside user. Use a to_chat instead, the people around him also dont need this feedback.